src/NuGet.Clients/NuGet.PackageManagement.UI/Models/NuGetProjectUpgradeWindowModel.cs

```diff
@@ -64,134 +65,66 @@ private string ProjectName
 
         public IList<PackageDependencyInfo> PackageDependencyInfos { get; }
 
-        // Changing CollapseDependencies updates the list of included packages
-        public bool CollapseDependencies
-        {
-            get { return _collapseDependencies; }
-            set
-            {
-                _collapseDependencies = value;
-                OnPropertyChanged(nameof(DirectDependencies));
-                OnPropertyChanged(nameof(TransitiveDependencies));
-            }
-        }
-
-        public IEnumerable<string> Warnings
+        public IEnumerable<PackageIdentity> NotFoundPackages
         {
             get
             {
-                if (_warnings == null)
+                if(_notFoundPackages == null)
                 {
-                    InitPackageUpgradeIssues();
-                }
-                return _warnings;
-            }
-        }
-
-        public IEnumerable<string> Errors
-        {
-            get
-            {
-                if (_errors == null)
-                {
-                    InitPackageUpgradeIssues();
-                }
-                return _errors;
-            }
-        }
-
-        public IList<PackageIdentity> NotFoundPackages
-        {
-            get
-            {
-                if (_notFoundPackages == null)
-                {
-                    InitPackageUpgradeIssues();
+                    GetUpgradeDependencyItems();
                 }
                 return _notFoundPackages;
             }
         }
 
-        public bool HasIssues => Errors.Any() || Warnings.Any();
-
-        public IEnumerable<NuGetProjectUpgradeDependencyItem> UpgradeDependencyItems
+        public ObservableCollection<NuGetProjectUpgradeDependencyItem> UpgradeDependencyItems
             => _upgradeDependencyItems ?? (_upgradeDependencyItems = GetUpgradeDependencyItems());
 
-        public IEnumerable<string> DirectDependencies => CollapseDependencies ? IncludedCollapsedPackages : AllPackages;
+        public IEnumerable<NuGetProjectUpgradeDependencyItem> DirectDependencies => UpgradeDependencyItems
+                .Where(upgradeDependencyItem => !upgradeDependencyItem.DependingPackages.Any());
```

Comment:
DirectDependencies should also check for `PromoteToTopLevel` flag now to get the full list.